src/components/Builder.tsx

```diff
@@ -313,7 +317,34 @@ const Builder = () => {
     const updateStarknetGoParams = (currentParamsObj: {
       [key: string]: any;
     }) => {
-      return method.starknetGo; // TODO: Implement this
+      const regexPattern = /provider\.(\w+)\(([^)]*)\)/;
+      const codeSnippet = method.starknetGo;
+
+      const updatedCode = codeSnippet.replace(
+        regexPattern,
+        (match, methodName, params) => {
+          const values = Object.entries(currentParamsObj).flatMap(
+            ([key, value]) => {
+              // if (key === "")
+              if (typeof value === "object" && !Array.isArray(value)) {
+                // If value is an object, return its stringified values
+                return Object.values(value).map((val) =>
+                  typeof val === "string" ? `"${val}"` : val
+                );
+              } else if (typeof value === "string") {
+                // If value is a string, return it with quotes
+                return `"${value}"`;
+              }
+              return value; // Return other types (like numbers) as is
+            }
+          );
+
+          let stringifiedParams = values.join(", ");
+          return `provider.${methodName}(${stringifiedParams})`;
+        }
+      );
+
+      return updatedCode;
     };
```

Comment:
I just implemented this function. Please rebase so we use that instead

Reply:
@jelilat there are a lot og merge conflicts in this branch now, maybe i'll close this pull request and open another with the suggested changes? Please let me know if this is fine.